railties/test/generators/app_generator_test.rb

```diff
@@ -296,6 +296,19 @@ def test_app_update_does_not_remove_rack_cors_if_already_present
     end
   end
 
+  def test_app_update_does_not_generate_assets_initializer_when_skip_sprockets_is_given
+    app_root = File.join(destination_root, "myapp")
+    run_generator [app_root, "--skip-sprockets"]
+
+    stub_rails_application(app_root) do
+      generator = Rails::Generators::AppGenerator.new ["rails"], { update: true, skip_sprockets: true }, { destination_root: app_root, shell: @shell }
```

Comment:
You should be using the same thing as the test below that is the `app:update` task. Can you change it?

Reply:
@rafaelfranca Done https://github.com/rails/rails/pull/32780/commits/c6c55d6a3ff89eda1039a8c0247e3f8b8e70130c , but it seems to stuck the travis job... Now I'm investigating the the cause.

Reply:
Cool! Yeah it is stuck. Try to run it locally to see.

Reply:
@rafaelfranca I don't know why, but it seems to load sprockets even if it's skipped.

To investigating the the cause, I have change the test as follows and ran `$ bin/test test/generators/app_generator_test.rb` .

```
   def test_app_update_does_not_generate_assets_initializer_when_skip_sprockets_is_given
     app_root = File.join(destination_root, "myapp")
-    run_generator [app_root, "--skip-sprockets"]
+    run_generator [app_root, "--skip-sprockets", "--skip-bootsnap"]

     FileUtils.cd(app_root) do
-      quietly { system("bin/rails app:update") }
+      system('cat config/application.rb')
+      system("bin/rails r 'p Object.constants.grep(/Sprockets/)'")
+      system("bin/rails r 'p Bundler.require(*Rails.groups).map(&:name).grep(/spro/)'")
+      system("bin/rails app:update")
     end
```

The output is below:

```
(snip)
...
require_relative 'boot'

require "rails"
# Pick the frameworks you want:
require "active_model/railtie"
require "active_job/railtie"
require "active_record/railtie"
require "active_storage/engine"
require "action_controller/railtie"
require "action_mailer/railtie"
require "action_view/railtie"
require "action_cable/engine"
# require "sprockets/railtie"
require "rails/test_unit/railtie"

# Require the gems listed in Gemfile, including any gems
# you've limited to :test, :development, or :production.
Bundler.require(*Rails.groups)

module Myapp
  class Application < Rails::Application
    # Initialize configuration defaults for originally generated Rails version.
    config.load_defaults 6.0

    # Settings in config/environments/* take precedence over those specified here.
    # Application configuration can go into files in config/initializers
    # -- all .rb files in that directory are automatically loaded after loading
    # the framework and any gems in your application.
  end
end
[:Sprockets]
["sprockets-export"]
    conflict  config/boot.rb
Overwrite /Users/tanimichi.tsukuru/ghq/github.com/ttanimichi/rails/railties/test/fixtures/tmp/myapp/config/boot.rb? (enter "h" for help) [Ynaqdh]
```

Reply:
This happens because `ENV['BUNDLE_GEMFILE']` has been specified during test execution. 
In `ENV['BUNDLE_GEMFILE']`, Rails's own Gemfile is specified instead of the generated application. In that file `sprockets-rails` is loaded and can not check correctly. 
(I think that the existing tests just passed by chance, and it was not tested correctly)

I think that some improvement is necessary for these, but I think that there is no problem in the way that the original `AppGenerator` is directly executed in this PR.

Reply:
@y-yagi Thank you for the information!

@rafaelfranca Can I remove the commit https://github.com/rails/rails/commit/c6c55d6a3ff89eda1039a8c0247e3f8b8e70130c and make this PR back to the original?

Reply:
@rafaelfranca  Or, how about this? https://github.com/rails/rails/pull/32780/commits/c842185e6513ab28b8e186337ce1ee4f8a596863
By using `switch_env` to set `ENV['BUNDLE_GEMFILE']` as nil, we can use the generated app's Gemfile to test the `app:update` task.

Reply:
No, this has not been tested correctly. 

Although trying to use application's Gemfile, gem of version `6.0.0.alpha` can not be installed, resulting in an error. Errors can be confirmed by deleting `quietly`. 

```diff 
+  def test_app_update_does_not_generate_assets_initializer_when_skip_sprockets_is_given
+    app_root = File.join(destination_root, "myapp")
+    run_generator [app_root, "--skip-sprockets"]
+
+    FileUtils.cd(app_root) do
+      switch_env "BUNDLE_GEMFILE", nil do
+        system("bin/rails app:update")
+      end
+    end
+
+    assert_no_file "#{app_root}/config/initializers/assets.rb"
+  end
```

```
$ ./bin/test -w test/generators/app_generator_test.rb:320
Run options: --seed 41189

# Running:

Could not find gem 'rails (~> 6.0.0.alpha)' in any of the gem sources listed in your Gemfile.
Run `bundle install` to install missing gems.
.
```  

As I mentioned earlier, can you restore the test and sqush the commit to one?

Reply:
Oh, I understand... Sorry, I didn't notice that. How difficult it is to test railties correctly!
And, we should probably add assertions of the return values of `system` in most of the railties' tests to prevent such cases... There may be more tests which are not tested correctly.

Reply:
> can you restore the test and sqush the commit to one?

@y-yagi  Done https://github.com/rails/rails/pull/32780/commits/59c3d539ab9d3c00d910939f63eaadb67aea719b